swagger.yaml

```diff
@@ -5193,6 +5203,18 @@ definitions:
       finalizing_stage_percentage:
         type: integer
 
+  disk-encryption:
+    type: object
+    properties:
+      enable_on:
+        type: string
+        description: Enable/disable disk encryption on master nodes, worker nodes, or all nodes.
+        enum: ['none', 'all', 'masters', 'workers']
```

Comment:
This configuration per node pool is nice.

How should this work if there are more disk partitions than just a singular root partition; does this need to have an additional opt-in/opt-out per disk partition?  Or even a unique disk encryption mode per disk partition?

Reply:
From (openshift docs)[https://docs.openshift.com/container-platform/4.7/installing/install_config/installing-customizing.html#installation-special-config-encrypt-disk-tpm2_installing-customizing]:
```
If you configure additional data partitions, they will not be encrypted unless encryption is explicitly requested
```

I guess that we will need additional objects in `luks` list in that case, we can have a different task for it.
WDYT @avishayt ?

Reply:
I assumed that the configuration in the docs was the suggested one for OpenShift and would encrypt the relevant partitions.  In what case would additional partitions be created?

Reply:
@lack? Can you answer @avishayt 's question?

Reply:
I'm trying to figure out the specific needs of Telco, where there may be a need for a separate partition set up for the exclusive use of the LocalStorageOperator.  I haven't yet gotten a straight answer about whether that's needed or not, though.

But a read of the luks section of the ignition spec makes it clear that if there are any additional partitions besides the root (a configuration that is supported by OpenShift I think), they each need their own luks section too if encryption is desired.

If these multiple partitions are something AssistedService allows users to set up, the encryption should be as flexible as the partition definition.

Reply:
> If these multiple partitions are something AssistedService allows users to set up

Do we @avishayt?

Anyway, I think this PR is big enough, we have a different task for disk partitioning https://issues.redhat.com/browse/MGMT-7461.